packages/block-editor/src/hooks/use-bindings-attributes.js

```diff
@@ -36,8 +36,7 @@ const createEditFunctionWithBindingsAttribute = () =>
 			const { getBlockBindingsSource } = unlock(
 				useSelect( blockEditorStore )
 			);
-			const { getBlockAttributes, updateBlockAttributes } =
-				useSelect( blockEditorStore );
+			const { getBlockAttributes } = useSelect( blockEditorStore );
 
 			const updatedAttributes = getBlockAttributes( clientId );
```

Comment:
Why are we calling a selector during render here? Btw I think we have access to the attributes through `props.attributes`, so we can shallow clone that?

Reply:
Btw down below we are mutating the attributes object directly in the store? 🤔 

```
updatedAttributes[ attributeName ] =
```

Reply:
Thanks for sharing the feedback! Actually, I wanted to review that part because in a new experiment I realized it could be improved as you suggest: [link](https://github.com/WordPress/gutenberg/blob/6baa85fab115301d3d80e9a79e8ee68f84603dd6/packages/block-editor/src/hooks/use-bindings-attributes.js#L53-L97). There, I'm getting the data via `props` and mutating the attributes in a different way.

Do you think that approach would be better?

Reply:
Sweet! Can you share the PR link? We can continue our discussion there.

Reply:
Sorry I forgot that! Here is the PR: [link](https://github.com/WordPress/gutenberg/pull/58723).

It is just an initial experiment that hasn't been tested and needs to be improved. I was considering reusing just the `attributesWithBindings` part and creating another pull request only with that if we consider it better than the current approach.